multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/signature/jpa/JpaSignatureKeyDao.java

```diff
@@ -53,6 +54,12 @@ public SignatureKeyPairImpl create(SignatureKeyPairImpl keyPair)
     requireNonNull(keyPair, "Required non-null key pair");
     try {
       doCreate(keyPair);
+    } catch (IntegrityConstraintViolationException x) {
+      throw new ConflictException(
+          format(
+              "Workspace with id '%s' referenced by signature key pair doesn't exist",
```

Comment:
I think the following formulating would be simpler
```
Unable to create signature key pair because referenced workspace with id 'workspaceb5lfsypy1st3eclz' doesn't exist
```